### DISCUSSION TITLE
references count?!

### DISCUSSION THREAD

**Discussion by amirongit** [microsoft/pyright#6440] [Ideas]
has it been discussed?
since pyright provides "references" functionality, is it hard to add a ref count?
consider this a feature request

extra links: https://github.com/fannheyward/coc-pyright/issues/1020

**Comment by erictraut** [comment]
I'm not sure what you mean by "references count". Are you looking for some way to display the total number of references without listing them individually? 

It's up to your client to display references that are returned by the language server. It could display them as a count if it wants, but that's outside the control of the language server.

**Reply by amirongit** [reply]
> Are you looking for some way to display the total number of references without listing them individually

yes; that's what I'm looking for.

> t's up to your client to display references that are returned by the language server. It could display them as a count if it wants, but that's outside the control of the language server.

ok then, thanks.

**Reply by fannheyward** [reply]
> references that are returned by the language server

@erictraut sorry to boring you, but do you sure Pyright provides this by `codeLens`? I traced Pyright logs, it didn't support `codeLensProvider`. Some language servers, for example rust-analzyer, will response references count for codeLens request, then client display them like bellow, this is what OP requests:

<img width="463" alt="image" src="https://github.com/microsoft/pyright/assets/345274/c27bc7b3-f4f3-49cf-9802-2127598f49bd">

**Reply by erictraut** [reply]
Thanks, I wasn't aware of that language server interface. I don't think Pylance implements that currently. If you're interested in having this added to Pylance, you could file an enhancement request for the Pylance team. As you're probably aware, we don't plan to add any new language server functionality in pyright, which is focused on static type checking functionality.

**Comment by amirongit** [comment]
closing
